plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveAnalyzeCorruptStatistics.java

```diff
@@ -78,6 +78,37 @@ private void prepareBrokenColumnStatisticsTable(String tableName)
         assertEquals(onMetastore("SELECT COUNT(1) FROM TAB_COL_STATS WHERE db_name = 'tpch' AND table_name = '" + tableName + "'"), "2");
     }
 
+    @Test
+    public void testAnalyzeCorruptPartitionStatisticsOnEmptyTable()
+    {
+        String tableName = "test_analyze_corrupt_partition_statistics_" + randomNameSuffix();
+
+        // Concurrent ANALYZE statements generate duplicated rows in Thrift metastore's PART_COL_STATS table when partition statistics is empty
+        prepareBrokenPartitionStatisticsTable(tableName);
+
+        // SHOW STATS should succeed even when the partition statistics is broken
+        assertQuerySucceeds("SHOW STATS FOR " + tableName);
+
+        // ANALYZE succeeds unlike TAB_COL_STATS, but drop_stats fails for tables having broken partition statistics
+        assertUpdate("ANALYZE " + tableName, 1);
+        assertThatThrownBy(() -> query("CALL system.drop_stats('tpch', '" + tableName + "')"))
```

Comment:
It seems that this problem was found on other query engines as well, but not addressed on the hive metastore side.

https://issues.apache.org/jira/browse/IMPALA-1976